pkg/csi/cinder/controllerserver.go

```diff
@@ -147,6 +164,38 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	return getCreateVolumeResponse(vol, ignoreVolumeAZ, req.GetAccessibilityRequirements()), nil
 }
 
+func (cs *controllerServer) getSnapshot(cloud openstack.IOpenStack, snapshotID string) (error, error) {
+	snap, err := cloud.GetSnapshotByID(snapshotID)
```

Comment:
I don't see a reason in returning two errors. Can you return a single error? Current code is very complex and hard to read.